### PR TITLE
Feature/templates tests

### PR DESCRIPTION
**Pull request by alexrrrr**
Add test infrastructure project and extract route config to this project.
Add tests for controllers.
Add tests for routes and mappers.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/161%23discussion_r63398751%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/161%23discussion_r63398834%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/161%23discussion_r63399052%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/161%23discussion_r63399079%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20cf807b2736a003792b715ac7bda485647f164e0e%20DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestExtensions/MapperTests.cs%2012%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/161%23discussion_r63398751%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Too%20general%20name%20for%20class%20name.%22%2C%20%22created_at%22%3A%20%222016-05-16T18%3A15%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestExtensions/MapperTests.cs%3AL1-90%22%7D%2C%20%22Pull%20cf807b2736a003792b715ac7bda485647f164e0e%20DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestForControllers/TemplatesExternalControllerTests.cs%20371%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/161%23discussion_r63399079%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Same.%22%2C%20%22created_at%22%3A%20%222016-05-16T18%3A17%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestForControllers/TemplatesExternalControllerTests.cs%3AL1-435%22%7D%2C%20%22Pull%20cf807b2736a003792b715ac7bda485647f164e0e%20DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestExtensions/MapperTests.cs%2075%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/161%23discussion_r63398834%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Nice%20code%21%21%21%20I%20like%20such%20code.%22%2C%20%22created_at%22%3A%20%222016-05-16T18%3A16%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestExtensions/MapperTests.cs%3AL1-90%22%7D%2C%20%22Pull%20cf807b2736a003792b715ac7bda485647f164e0e%20DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestForControllers/TemplatesExternalControllerTests.cs%20299%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/161%23discussion_r63399052%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22You%20can%20inline%20this%20status%20code.%22%2C%20%22created_at%22%3A%20%222016-05-16T18%3A17%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestForControllers/TemplatesExternalControllerTests.cs%3AL1-435%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull cf807b2736a003792b715ac7bda485647f164e0e DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestExtensions/MapperTests.cs 12'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/161#discussion_r63398751'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestExtensions/MapperTests.cs:L1-90</a></b>
- <a href='https://github.com/TerraVenil'></a> Too general name for class name.
- [ ] <a href='#crh-comment-Pull cf807b2736a003792b715ac7bda485647f164e0e DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestExtensions/MapperTests.cs 75'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/161#discussion_r63398834'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestExtensions/MapperTests.cs:L1-90</a></b>
- <a href='https://github.com/TerraVenil'></a> Nice code!!! I like such code.
- [ ] <a href='#crh-comment-Pull cf807b2736a003792b715ac7bda485647f164e0e DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestForControllers/TemplatesExternalControllerTests.cs 299'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/161#discussion_r63399052'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestForControllers/TemplatesExternalControllerTests.cs:L1-435</a></b>
- <a href='https://github.com/TerraVenil'></a> You can inline this status code.
- [ ] <a href='#crh-comment-Pull cf807b2736a003792b715ac7bda485647f164e0e DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestForControllers/TemplatesExternalControllerTests.cs 371'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/161#discussion_r63399079'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestForControllers/TemplatesExternalControllerTests.cs:L1-435</a></b>
- <a href='https://github.com/TerraVenil'></a> Same.

<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/161?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/161?approve=1'></a>&nbsp;<a href='https://github.com/ITsvetkoFF/Kv-013/pull/161'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by ITsvetkoFF** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/ITsvetkoFF/Kv-013/pull/161'>Refresh</a>.<a href='#crh-update-none'></a>

File: DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestExtensions/MapperTests.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+using System.Diagnostics;
+using FluentAssertions;
+using GitHubExtension.Templates.CommunicationModels;
+using GitHubExtension.Templates.DAL.Model;
+using GitHubExtension.Templates.Mappers;
+using Xunit;
+
+namespace GitHubExtension.Templates.Tests.TestExtensions
+{
+    public class MapperTests
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestExtensions/MapperTests.cs]: Too general name for class name.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestExtensions/MapperTests.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+        [Theory]
+        [MemberData("DataForTemplatesModelTest")]
+        public void TemplatesModelTest(Template entityToTemplateModel, TemplatesModel expectedTemplatesModelModel)
+        {
+            // Act
+            TemplatesModel templatesModel = entityToTemplateModel.ToTemplateModel();
+
+            // Assert
+            templatesModel.ShouldBeEquivalentTo(expectedTemplatesModelModel);
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestExtensions/MapperTests.cs:75]: Nice code!!! I like such code.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestForControllers/TemplatesExternalControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+        [Theory]
+        [MemberData("DataForIssueCreatedResult")]
+        public async void ShoudReturnCreatedForCreateIssueTemplate(CreateUpdateTemplateModel model)
+        {
+            // Arrange
+            var templatesCommand = Substitute.For<ITemplatesCommand>();
+            templatesCommand.CreateTemplateAsync(model).ReturnsForAnyArgs(Task.FromResult(HttpStatusCode.Created));
+            TemplatesExternalController controller = new TemplatesExternalController(templatesCommand, Substitute.For<ITemplatesQuery>());
+            controller.SetUserForController(TestUserName, Claims);
+            var expectedResult = HttpStatusCode.Created;
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestForControllers/TemplatesExternalControllerTests.cs]: You can inline this status code.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestForControllers/TemplatesExternalControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+        [Theory]
+        [MemberData("DataForIssueOkResult")]
+        public async void ShoudReturnOkForIssueTemplate(CreateUpdateTemplateModel model)
+        {
+            // Arrange
+            var templatesCommand = Substitute.For<ITemplatesCommand>();
+            templatesCommand.UpdateTemplateAsync(model).ReturnsForAnyArgs(Task.FromResult(HttpStatusCode.OK));
+            TemplatesExternalController controller = new TemplatesExternalController(templatesCommand, Substitute.For<ITemplatesQuery>());
+            controller.SetUserForController(TestUserName, Claims);
+            var expectedResult = HttpStatusCode.OK;
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Templates.Tests/TestForControllers/TemplatesExternalControllerTests.cs]: Same.

**Pull request merged**